packages-content-model/roosterjs-content-model-core/lib/corePlugin/ContentModelCopyPastePlugin.ts

```diff
@@ -250,6 +256,41 @@ class ContentModelCopyPastePlugin implements PluginWithState<CopyPastePluginStat
     };
 }
 
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export function adjustSelectionForCopyCut(pasteModel: ContentModelDocument) {
+    let selectionMarker: ContentModelSelectionMarker | undefined;
+    let firstBlock: ContentModelBlock | undefined;
```

Comment:
You can declare firstBlock as ContentModelParagraph and only set is value when the block is Paragraph. Then no need to check at line 272

Reply:
Sounds good, but I think I will need to move the same check, because the callback block is ContentModelBlock, so to type check we still need to verify that the blockType is equal to Paragraph.